### ISSUE TITLE
[Java] A false negative about NP_NONNULL_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR

### ISSUE THREAD

**Issue by Zustin** [spotbugs/spotbugs#1642]
Hi, I found a false negative about the rule [NP_NONNULL_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#np-non-null-field-is-not-initialized-np-nonnull-field-not-initialized-in-constructor) when used SpotBugs.

The minimal case is below:

```java
@Nonnull
static Object obj;  // should report a warning because it is not initialized.
static {
    obj = obj;
}
```

Actually, the variable *obj* is not initialized. It is still a null object. The official test case is here [Link](https://github.com/spotbugs/spotbugs/blob/51e586bed98393e53559a38c1f9bd15f54514efa/spotbugsTestCases/src/java/bugPatterns/NP_NONNULL_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR.java)

Version: using IntelliJ SpotBugs plugin 1.2.4 with SpotBugs version 4.1.3

**Comment by KengoTODA** [comment]
released as 4.4.2
